Blockzilla/BrowserViewController.swift

```diff
@@ -480,39 +481,55 @@ class BrowserViewController: UIViewController {
             UIKeyCommand(input: "]", modifierFlags: .command, action: #selector(BrowserViewController.goForward), discoverabilityTitle: UIConstants.strings.browserForward),
         ]
     }
+
+    func canShowTrackerStatsShareButton() -> Bool {
+        return NSLocale.current.identifier == "en_US" && !AppInfo.isKlar
+    }
     
-    func shouldShowTrackerStatsShareButton() -> Bool {
-        let shouldShowTrackerStatsToUser = UserDefaults.standard.object(forKey: userDefaultsShareTrackerStatsKey) as! Bool?
+    func shouldShowTrackerStatsShareButton(percent: Int = 30, userDefaults:UserDefaults = UserDefaults.standard) -> Bool {
+        var shouldShowTrackerStatsToUser = userDefaults.object(forKey: BrowserViewController.userDefaultsShareTrackerStatsKeyNEW) as! Bool?
         
         if shouldShowTrackerStatsToUser == nil {
-            // User has not been put into a bucket for determining if it should be shown
-            // 10% chance they get put into the group that sees the share button
-            // arc4random_uniform(10) returns an integer 0 through 9 (inclusive)
-            if arc4random_uniform(10) == 0 {
-                UserDefaults.standard.set(true, forKey: userDefaultsShareTrackerStatsKey)
-
+            // Check to see if the user was previously opted into the experiment
+            shouldShowTrackerStatsToUser = userDefaults.object(forKey: BrowserViewController.userDefaultsShareTrackerStatsKeyOLD) as! Bool?
+            
+            if shouldShowTrackerStatsToUser != nil {
+                // Remove the old flag
+                userDefaults.removeObject(forKey: BrowserViewController.userDefaultsShareTrackerStatsKeyOLD)
+            }
+            
+            if shouldShowTrackerStatsToUser == true {
+                // User has already been opted into the experiment, continue showing the share button
+                userDefaults.set(true, forKey: BrowserViewController.userDefaultsShareTrackerStatsKeyNEW)
+                return true
             } else {
-                UserDefaults.standard.set(false, forKey: userDefaultsShareTrackerStatsKey)
-                return false
+                // User has not been put into a bucket for determining if it should be shown
+                // 30% chance they get put into the group that sees the share button
+                // arc4random_uniform(100) returns an integer 0 through 99 (inclusive)
+                if arc4random_uniform(100) < percent {
+                    userDefaults.set(true, forKey: BrowserViewController.userDefaultsShareTrackerStatsKeyNEW)
+                    shouldShowTrackerStatsToUser = true
+                } else {
+                    userDefaults.set(false, forKey: BrowserViewController.userDefaultsShareTrackerStatsKeyNEW)
+                    return false
+                }
             }
         }
         
         if shouldShowTrackerStatsToUser == false ||
```

Comment:
this will change to 'if canShowTrackerStatsShareButton == false' after the refactoring above.